### ISSUE TITLE
Line stepping bug due to mishandling indirect call

### ISSUE THREAD

**Issue by mistymntncop** [EpicGamesExt/raddebugger#322]
~I'm currently on commit [23da2f8].~ On alpha 0.9.12.
I found an interesting bug using the following test code. 

```
#include <stdio.h>
#include <stdint.h>
#include <windows.h>
 
uint8_t *code = 0;

uint64_t clear_rax() {
    return 0;
}

void c(void) {
    clear_rax();
    ((void ( *)()) code)();
}

int main(void) {
    printf("1\n");
 
    uint8_t *mem = VirtualAlloc(NULL, 0x1000, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
    uint8_t code_vals[] = {
        0x55,                        //push   rbp
        0x48, 0x89, 0xE5,            //mov    rbp,rsp
        0xCC,                        //int3
        0x48, 0x89, 0xEC,            //mov    rsp,rbp
        0x5D,                        //pop    rbp
        0xC3                         //ret 
    };
    code = mem;
    memcpy(code, code_vals, sizeof(code_vals));
        
    c();

    printf("2\n");
    return 0;
}
```

To reproduce:
1. place a line breakpoint on function c(). Then run til you hit the breakpoint
![image](https://github.com/user-attachments/assets/8960dbb2-74d6-4755-9f13-5b237eb8c23e)

2. Line step until instruction pointer is on the indirect call instruction
![image](https://github.com/user-attachments/assets/0fdd39ef-80ea-42b0-8e82-6ab990c09c2c)

3. Line step into (has to be in source view) into the call instruction. You will encounter this crash.
![image](https://github.com/user-attachments/assets/df897c3d-c649-4db6-8410-f848cc70a514)

In this example 0x00000232583c0000 (code) is the base of our VirtualAlloc allocation where we copied our instructions to. This is where the instruction pointer should be. Instead we are at 0x00000232583c00CC (offset +0xCC). As you know 0xCC is the breakpoint instruction. 
What I believe is happening is the debugger is writing the breakpoint instruction to the wrong location. 
E.g.  in our example 0x00007FF65C4740C8 is the address where our code pointer (0x00000232583c0000) resides. The debugger is overwriting the lowest byte of 0x00000232583c0000 to be 0x00000232583c00CC (and resetting it after the exception is hit). More simply put its as if the debugger is treating the instruction "call [0x00007FF65C4740C8]" like "call 0x00007FF65C4740C8" instead.

I think this is from the code involving DASM_CtrlFlowInfo in ["d_trap_net_from_thread__step_into_line"](https://github.com/EpicGamesExt/raddebugger/blob/6a1c36b21298899e2c4067a256063370079c8784/src/dbg_engine/dbg_engine_core.c#L598C4-L598C5) (dev branch) not taking this into account. As the "jump_dest_vaddr" seems to be the address used by the indirect call (should be the pointer at that address instead I think).

This probably affects indirect jmp too.

Thanks.

**Comment by mistymntncop** [comment]
My bad. Seems to be working on 0.9.11 actually. Apologies.

**Comment by mistymntncop** [comment]
Ok. I am reopening this issue as it seems to affect alpha 0.9.12 [0x94fb505]. The "call [1234]" should be rip relative instructions I think?

**Comment by ryanfleury** [comment]
Thanks for thoroughly investigating this, it was very helpful. This should now be fixed as of bcd16a4f29d91f7c71d32e2451b43ebd77726c32 - I was incorrectly using the operand address as a jump destination, as you suggested.
